homeassistant/components/reolink/host.py

```diff
@@ -162,4 +179,140 @@ async def disconnect(self):
 
     async def stop(self, event=None):
         """Disconnect the API."""
+        await self.unregister_webhook()
         await self.disconnect()
+
+    async def subscribe(self) -> bool:
+        """Subscribe to motion events and register the webhook as a callback."""
+        if self.webhook_id is None:
+            if not self.register_webhook():
+                return False
+
+        if self._api.subscribed:
+            _LOGGER.debug(
+                "Host %s: is already subscribed to webhook %s",
+                self._api.host,
+                self._webhook_url,
+            )
+            return True
+
+        if await self._api.subscribe(self._webhook_url):
+            _LOGGER.info(
+                "Host %s: subscribed successfully to webhook %s",
+                self._api.host,
+                self._webhook_url,
+            )
+        else:
+            _LOGGER.error("Host %s: webhook subscription failed", self._api.host)
```

Comment:
Please don't log errors when we're going to raise `ConfigEntryNotReady`. That is going to spam the log when we retry the config entry setup.

Instead of using booleans for flow control I suggest raising an internal exception and passing the error message as the exception argument. Create a custom exception for each reason and inherit `HomeAssistantError`. Then catch the exceptions in config entry setup and raise `ConfigEntryNotReady` from the exception.